Description: Fix for building with libkdcraw shipped with KDE 4.10
commit ee9e93ba5be584c52d404a9c3b534cecaf11febd
Author: Rex Dieter <rdieter@math.unl.edu>
Date:   Tue Dec 4 23:21:21 2012 -0600

    fix build for newer libkdcraw api
    
    CCBUG: 307148

Origin: upstream
Forwarded: not-needed
Reviewed-By: Scott Kitterman <scott@kitterman.com>
Last-Update: 2012-12-09

Index: git/ImageManager/RawImageDecoder.cpp
===================================================================
--- git.orig/ImageManager/RawImageDecoder.cpp	2013-01-30 18:27:13.000000000 +0100
+++ git/ImageManager/RawImageDecoder.cpp	2013-01-30 18:27:21.000000000 +0100
@@ -24,6 +24,7 @@
 #ifdef HAVE_KDCRAW
 #  include <libkdcraw/kdcraw.h>
 #  include <libkdcraw/rawfiles.h>
+#  include <libkdcraw/version.h>
 #endif
 #include <kdebug.h>
 #include <DB/FileName.h>
@@ -37,7 +38,11 @@
     Q_UNUSED( dim );
 
 #ifdef HAVE_KDCRAW
+#if KDCRAW_VERSION >= 0x020200
+    if ( !KDcrawIface::KDcraw::loadRawPreview( *img, imageFile.absolute() ) )
+#else
     if ( !KDcrawIface::KDcraw::loadDcrawPreview( *img, imageFile.absolute() ) )
+#endif
         return false;
 
     if ( Settings::SettingsData::instance()->useRawThumbnail() &&
